appserver/ejb/ejb-container/src/main/java/com/sun/ejb/codegen/EjbOptionalIntfGenerator.java

```diff
@@ -265,157 +213,154 @@ public void generateSubclass(Class superClass, String subClassName, Class delega
                     boolean isStatic = Modifier.isStatic(modifiers);
 
                     if( (isPackage || isProtected) && !isStatic ) {
-                        generateNonAccessibleMethod(tv, mth);
+                        generateNonAccessibleMethod(cw, method);
                     }
-                    allMethods.add(mth);
+                    allMethods.add(method);
                 }
             }
         }
 
         // add toString() method if it was not overridden
-        java.lang.reflect.Method mth = Object.class.getDeclaredMethod("toString");
-        if (!hasSameSignatureAsExisting(mth, allMethods)) {
-            generateToStringBeanMethod(tv, superClass);
+        try {
+            Method method = Object.class.getDeclaredMethod("toString");
+            if (!hasSameSignatureAsExisting(method, allMethods)) {
+                generateToStringBeanMethod(cw, superClass);
+            }
+        } catch (NoSuchMethodException ignored) {
+            // Should never be reached
```

Comment:
If it should never be reached, then better throw IllegalStateException, but don't hide it.